src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,159 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
     @Override
     public void add(T value) {
+        linkLast(value);
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            linkLast(value);
+        } else {
+            linkBefore(value, getNode(index));
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element : list) {
+            this.add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return getNode(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> node = getNode(index);
+        T oldVal = node.item;
+        node.item = value;
+        return oldVal;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        return unlink(getNode(index));
     }
 
     @Override
     public boolean remove(T object) {
+        for (Node<T> node = first; node != null; node = node.next) {
+            if ((object != null && object.equals(node.item))
+                    || (object == null && node.item == null)) {
```

Comment:
```suggestion
            if (object == node.item || object != null && object.equals(node.item)) {
```

Reply:
> 

Hi,
I figured out the mistake I made when tried to fix this:

if ((object != null && object.equals(node.item))
                    || (object == null && node.item == null)) {...}

to this:

if (object == node.item || object != null && object.equals(node.item)) {...}

Thanks!